### ISSUE TITLE
XML Library: Double namespace during Element To String

### ISSUE THREAD

**Issue by CosmaCLI** [robotframework/robotframework#4999]
Hi everyone!

I'm using the XML Lib with lxml
```
Library     XML    use_lxml=True
```
In my Teststep I'm loading a file and convert it to a XML:

```
    ${my_config_string}=    OperatingSystem.Get File
    ...    path_to_file
    ...    encoding=UTF-8
    ${my_config_xml}=    parse xml    ${my_config_string}
```
and its working fine - the XML looks like expected.

```
<config xmlns="http://tail-f.com/ns/config/1.0">
  <devices xmlns="http://tail-f.com/ns/ncs">
    <device>
.... 
```
But as soon as I use the Element To String Keyword the namespaces are doubled.
```
    ${my_config_candidate}=    Element To String
    ...    ${my_config_xml}
    ...    xpath=.
    ...    encoding=UTF-8
```

Result:
```
<config xmlns="http://tail-f.com/ns/config/1.0" xmlns="http://tail-f.com/ns/config/1.0">
  <devices xmlns="http://tail-f.com/ns/ncs" xmlns="http://tail-f.com/ns/ncs">
    <device>
```

It's not happening if I load the Library without lxml 

Thank you very much
BR
Christina

**Comment by pekkaklarck** [comment]
I was able to reproduce the issue. This is why it happens:
- ElementTree stoes namespaces to tag names using [Clark notation](http://www.jclark.com/xml/xmlns.htm) so that, for example, `<name xmlns="uri"/>` is represented as `<{uri}nane/>`.
- Clark notation is pretty annoying to use with xpath expressions and the XML library, by default strips it. It removes the `{uri}` part and adds the uri to `xmlns` attribute.
- Stripping Clarks notation causes problems with `lxml.etree` when elements are saved.
- We avoid the above problem adding `{uri}` back and removing the `xmlns` attribute when elements are saved to files.
- The above workaround isn't done when converting an element to a string. I already tested that doing that fixes the issue.

I haven't tested do duplicate `xmlns` attribute cause concrete problems or not, but the same issue certainly causes problems when using namespaces with prefixes. Luckily the aforementioned fix fixes that issue as well.

**Comment by pekkaklarck** [comment]
The fix just made it to RF 7.0. I plan to create release candidate 3 on Monday and we hopefully get the final release out already on Wednesday. It would be great if you @CosmaCLI could test the fix once the release candidate with the fix is out.

**Comment by CosmaCLI** [comment]
Forgot to answer here - yes it works fine now. Thank you!
